content/blog/2024-06-18-gsoc-uefi-gop.mdx

```diff
@@ -0,0 +1,43 @@
+---
+title: "GSoC'24: UEFI Graphics Output Protocol Support in Unikraft"
+description: |
+  This project aims to add support for the modern UEFI GOP interface in Unikraft.
+publishedDate: 2024-06-18
+authors:
+- Sriprad Potukuchi
+tags:
+- gsoc
+- gsoc24
+- uefi
+- booting
+---
+
+## Project Overview
+<img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-generic-badge.svg" align="right" />
+
+Unikraft's ability to print logs to the screen is quite limited, which makes it harder to debug it during development. While it is entirely possible to use the serial port console, it is not as easily accessible or available on usual consumer machines.
+
+The widely available and standardized [UEFI Graphics Output Protocol](https://uefi.org/specs/UEFI/2.10/12_Protocols_Console_Support.html#efi-graphics-output-protocol) (GOP) interface is an excellent candidate. It provides a raw framebuffer, which can be used to set individual pixels on the screen.
+
+This project aims to implement a console based on UEFI GOP, which is capable of parsing fonts and displaying text on the screen by setting the appropriate pixels in the framebuffer.
+
+## Progress
+- I have implemented a low level driver that locates the GOP protocol and gets the framebuffer (and some metadata like width and height).
+
+- I've created a [convenience script](https://github.com/procub3r/unikraft/blob/poc/run.sh) that helps build a UEFI image and run it in qemu.
+
+- I have also written a small [proof of concept](https://github.com/procub3r/unikraft/tree/poc) that demonstrates our ability to control individual pixels on the screen:
+
+  <Image
+        src="/images/uefi-gop-proof-of-concept.png"
+        title="UEFI GOP Proof of Concept"
+        description="A gradient displayed on the screen (in qemu-system-x86_64)"
+  />
+
+## Next Steps
+- Display characters from a bitmap font.
```

Comment:
What font have you chosen to start with?

Reply:
https://github.com/dhepper/font8x8 looks cool. I haven't set it in stone yet though.

Reply:
"License: Public Domain" perfect :+1: 
What I had in mind was the 16x8 VGA font e.g. <https://www.inp.nsk.su./~bolkhov/files/fonts/univga/>
But if this 8x8 is clear enough, I would like it more as well :smile: .

Sure, let's give this 8x8 a shot.